README.md

```diff
@@ -1,25 +1,172 @@
-## Proxy API
-This app is proxy for developer API with testing and learning purposes.
+# Riot Developer Example App
+This project is a consumption example of [Riot Games Developer API](https://developer.riotgames.com/apis) (RGAPI).
 
-### How to use?
-Any request made to `localhost:${APP_PORT}/*` will be proxied to RIOT Developer api (the region is defined at the environment of this proxy `RIOT_API_BASE_URI`).
+All endpoints requires authentication and there are two ways of authentication:
 
-Only HTTP Methods and request body will be proxied at request and response.
+## API Keys
+Anyone who signs into the developer portal will automatically be granted an API key that will be associated with their account. Your API key allows you to start using the API immediately, and allows us to generate API usage metrics for your key. [See example](#summoner-overview).
 
-## Hot Reload
-The dev docker container uses a tool called [CompileDaemon](https://github.com/githubnemo/CompileDaemon) to hot reload any changes made to the code.
+For more information about API Keys read the [full docs](https://developer.riotgames.com/docs/portal#web-apis_api-keys).
 
-## Setup
-To setup this app is needed `docker`, `docker-compose` and `make`.
+## RSO Authentication (OAUTH2)
+Some endpoints requires authorization from the user, and for this purpose RGAPI uses [oauth2](https://oauth.net/2/) protocol. [See example](#my-account).
 
-Steps:
-1. Create `.env` by running `cp .env.example .env`
-2. Set env var `RIOT_API_TOKEN` using your api credential
-3. Run `make up`
+# Project Setup
+To setup this project locally a few steps are needed, they are:
+- [Configure environment variables](#configure-environment-variables)
+- [Add rule for localhost domain (only for oauth2 authentication)](#add-rule-for-localhost-domain-only-for-oauth2-authentication)
+- [Boot Golang server](#boot-golang-server)
+
+## Configure environment variables
+### API Key
+You can get the API Key after sign in at [Riot Games Developer Portal](https://developer.riotgames.com/login). This token will be used as environment variable `RIOT_API_TOKEN`.
+
+### Get credentials for oauth2
+For oauth2 authentication are needed two credentials, they are `CLIENT_ID` and `CLIENT_SECRET` and will be used respectively for environment variables `RSO_CLIENT_ID` and `RSO_CLIENT_SECRET`.
+
+To get your credentials you need to [...]
```

Comment:
I need to understand how we are going to guide the creation oauth2 credentials . (If we are going to send everyone to email)

Maybe provide a boilerplate?

Reply:
Short-term we should probably add an email template to the docs part of the site that they can fill out. Long-term it will be a self-service guided flow in the developer portal

Reply:
Done!